src/bin/common/solving_loop.ml

```diff
@@ -429,21 +431,54 @@ let main () =
     | Sat model ->
       set_mode Sat ~model st
   in
-  (* The function In_channel.input_all is not available before OCaml 4.14. *)
-  let read_all ch =
-    let b = Buffer.create 113 in
-    try
-      while true do
-        Buffer.add_channel b ch 30
-      done;
-      assert false
-    with End_of_file ->
-      Buffer.contents b
+  (* Prepare the input source for Dolmen from an input source for Alt-Ergo. *)
+  let mk_files src =
+    let lang =
+      match Options.get_input_format () with
+      | Some Native -> Some Dl.Logic.Alt_ergo
+      | Some Smtlib2 version -> Some (Dl.Logic.Smtlib2 version)
+      | Some (Why3 | Unknown _) | None -> None
+    in
+    let filename, dir, path =
+      match src with
+      | `Stdin -> "<stdin>", "", ""
+      | `File path -> Filename.basename path, Filename.dirname path, path
+      | `Raw (filename, _) -> filename, "", ""
+    in
+    let is_incremental =
+      match lang with
+      | Some (Dl.Logic.Smtlib2 _) | None -> true
+      | _ -> false
+    in
+    let src =
+      match src with
+      | `Stdin when is_incremental ->
+        set_output_format ".smt2";
+        `Stdin
+      | `Stdin ->
+        `Raw (filename, Compat.In_channel.input_all stdin)
+      | `File path when Filename.check_suffix path ".zip" ->
+        Filename.(chop_extension path |> extension) |> set_output_format;
+        let content = AltErgoLib.My_zip.extract_zip_file path in
+        `Raw (Filename.chop_extension filename, content)
+      | `File path ->
+        Filename.extension path |> set_output_format;
+        let cin = open_in path in
+        let content = Compat.In_channel.input_all cin in
+        close_in cin;
+        `Raw (filename, content)
```

Comment:
Note to self: not sure why we had this, I think this can just be `` `File path`` (provided that we call `set_output_format` first)?